dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java

```diff
@@ -176,9 +175,31 @@ public boolean isNumeric()
         return NUMERIC_TYPES.contains( this );
     }
 
-    public boolean isAggregateable()
+    public boolean isAggregatable()
     {
-        return aggregateable;
+        return aggregatable;
+    }
+
+    public boolean isAggregatable( AggregationType aggregationType )
+    {
+        if ( !aggregationType.isAggregatable() || !isAggregatable() )
+        {
+            return false;
+        }
+
+        if ( this == FILE_RESOURCE && aggregationType != AggregationType.COUNT )
+        {
+            return false;
+        }
+
+        if ( (this == TEXT || this == LONG_TEXT || this == LETTER) )
+        {
+            return aggregationType == AggregationType.NONE;
+        }
+        else
+        {
+            return aggregationType != AggregationType.NONE;
+        }
     }
```

Comment:
Can we add a unit test to verify the behavior?

Reply:
Done

Reply:
Great.